### ISSUE TITLE
Installation Problem: [Yarn]

### ISSUE THREAD

**Issue by MITDD6338** [yarnpkg/yarn#5944]
**I Am Using Windows 10:**

**Yarn Install Does't Not Working :**

**Comment by kenshinji** [comment]
Could you pls paste the error message you encountered here? much appreciated.

**Comment by MITDD6338** [comment]
info fsevents@1.2.4: The platform "win32" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "react-native > eslint-plugin-react-native@3.2.1" has unmet peer dependency "eslint@^3.17.0 || ^4.0.0".
warning " > babel-jest@23.0.1" has unmet peer dependency "babel-core@^6.0.0 || ^7.0.0-0".
warning " > react-native-google-signin@0.12.0" has unmet peer dependency "prop-types@^15.5.10".

**Comment by MITDD6338** [comment]
I am Trubble The New Version Node.js And React-Native (0.55)  
My Old Project Are Not Working Properly Some Normal Problem Are Thread ...
So please Help ME!!!!!!!!!!

**Comment by rodrigooler** [comment]
@MITDD6338 could you print the output of your terminal?

**Comment by rodrigooler** [comment]
@MITDD6338 apparently it is installing everything normal and creating the file yarn.lock. Is the problem in running some script in package.json? Apparently no error is occurring in yarn through your print.

**Comment by MITDD6338** [comment]
{
  "name": "Test",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "eslint": "^4.19.1",
    "firebase": "^5.0.4",
    "react": "16.3.1",
    "react-native": "0.55.4",
    "react-native-google-signin": "^0.12.0"
  },
  "devDependencies": {
    "babel-jest": "23.0.1",
    "babel-preset-react-native": "4.0.0",
    "jest": "23.1.0",
    "react-test-renderer": "16.3.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
**This Package.json File**

**Comment by kenshinji** [comment]
@MITDD6338 From the screenshot you posted above, we can't see any error there. Why did you say that **Yarn Install Does't Not Working**?

**Comment by MITDD6338** [comment]
D:\OfficeWork\Live Project\Test>yarn add firebase
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\uuid.cmd as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\uuid
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\uuid as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\uuid
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\jsesc.cmd as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\jsesc
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\jsesc as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\jsesc
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\esparse.cmd as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\esprima
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\esparse as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\esprima
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\esvalidate.cmd as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\esprima
npm WARN rm not removing D:\OfficeWork\Live Project\Test\node_modules\.bin\esvalidate as it wasn't installed by D:\OfficeWork\Live Project\Test\node_modules\esprima

> grpc@1.11.3 install D:\OfficeWork\Live Project\Test\node_modules\grpc
> node-pre-gyp install --fallback-to-build --library=static_library

internal/modules/cjs/loader.js:596
    throw err;
    ^

Error: Cannot find module 'D:\OfficeWork\Live Project\Test\node_modules\node-pre-gyp\bin\node-pre-gyp'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:594:15)
    at Function.Module._load (internal/modules/cjs/loader.js:520:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:744:10)
    at startup (internal/bootstrap/node.js:238:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:572:3)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! grpc@1.11.3 install: `node-pre-gyp install --fallback-to-build --library=static_library`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the grpc@1.11.3 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Admin\AppData\Roaming\npm-cache\_logs\2018-06-07T07_24_56_144Z-debug.log

D:\OfficeWork\Live Project\Test>


This Error Generate Using Yarn 
I am Search Problem Yarn Again Install This Above Screenshot  yarn Not Working 
And pakage.Json File Not Working My Pc

**Comment by PriyankaVrma** [comment]
<img width="592" alt="yarn test" src="https://user-images.githubusercontent.com/39528291/41279958-b3d9ec94-6e25-11e8-8fca-e2fa67bd5d29.png">

Can you please provide me the solution for this ....

**Comment by rodrigooler** [comment]
@PriyankaVrma how did you install? You could use choco install yarn which is the safest way to install on windows systems.

**Comment by vrobinson** [comment]
I am also seeing this error. My configuration is Windows 10, node.js v10.5.0, and yarn 1.7.0 and I am trying to run: yarn add gulp@4 --dev on a new project.

**Comment by vrobinson** [comment]
FYI: I still see the error after upgrading to node v10.6.0.

**Comment by vrobinson** [comment]
Still getting this warning on node v10.8.0 and yarn v1.9.4.

I have noticed something else which may be useful: Not all packages produce the warning.

If I start with a clean project and add the ramda package I do not get the warning, but once I add gulp@4 it appears. If I reverse the order by adding gulp@4 first and add ramda second, then the addition of ramda also produces the warning as does adding any other package (it seems). I assume this is because yarn is validating that all existing packages are up to date and it is re-discovering the problem with gulp@v4 each time.

I get the same behavior with or without the --dev switch.

Furthermore, if I try the same thing with npm I also get the warning:

```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
```

It exhibits the same behavior as yarn when reversing the installation order: ramda (no warning), gulp@4 (warning) and in reverse: gulp@4 (warning), ramda (warning).

There is a gulp issue for this warning also and they have elected not to fix it:
https://github.com/gulpjs/gulp/issues/1810

There is another issue addressing this 'warning' for yarn and it was resolved by changing the 'warning' to an 'info':
https://github.com/yarnpkg/yarn/issues/3738

However, in my opinion the 'info' message still sounds like a warning and causes people like me to waste time searching for a resolution so they don't have to see this output. A message similar to the following would be better (also note that it should be a single line output instead of two lines):

    Skipping platform specific dependency 'fsevents@1.2.4' because it is not required on the 'win32' platform.

Furthermore, I only want to see this warning when installing a package that has the dependency on 'fsevents'. I do NOT want to see it when I am installing ramda or anything else that does not use 'fsevents'.

It would also be even better if there were a way to silence/hide this specific warning (and perhaps others). Perhaps a way to tell yarn to always ignore this specific package for a given project or perhaps a command line switch for the add command to hide the message. Another possibility would be to only expose the message to developers with a direct dependency and silence it if the dependency is indirect.
